logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/auth/ValidateUserHandleUseCase.kt

```diff
@@ -1,24 +1,21 @@
 package com.wire.kalium.logic.feature.auth
 
 interface ValidateUserHandleUseCase {
-    operator fun invoke(handle: String): Boolean
+    operator fun invoke(handle: String): ValidateUserHandleResult
 }
 class ValidateUserHandleUseCaseImpl: ValidateUserHandleUseCase  {
-    override operator fun invoke(handle: String): Boolean {
-        return when {
-            isHandleTooShort(handle) -> false
-            !validateHandle(handle) -> false
-            else -> true
-        }
+    override operator fun invoke(handle: String): ValidateUserHandleResult {
+        val handleWithoutInvalidChars = handle.replace(Regex(HANDLE_FORBIDDEN_CHARACTERS_REGEX), "")
+        val hasValidChars = handle == handleWithoutInvalidChars
+        val hasValidLength = handleWithoutInvalidChars.length in HANDLE_MIN_LENGTH..HANDLE_MAX_LENGTH
+        return ValidateUserHandleResult(hasValidChars && hasValidLength, handleWithoutInvalidChars)
```

Comment:
If we do the check on every char entered by the user, `hasValidChars && hasValidLength` will always be false  for the first char, and it will trigger the shake animation, is this the intended behavior?

Reply:
For the input I just wanted to compare the text that the user entered with `handleWithoutInvalidChars` and if both aren't the same then it means that the "shake" animation should be started, but maybe you're right and that's not clear enough, so I changed the `ValidateUserHandleResult` to be sealed class and thus be more versatile.